pkg/nodeagent/registry/remote_extractor.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"archive/tar"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	containerregistryv1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+type remoteExtractor struct{}
+
+func (remoteExtractor) ExtractFromLayer(image, pathSuffix, dest string) error {
+	// In the local environment, we pull Gardener images built via skaffold from the local registry running in the kind
+	// cluster. However, on local machine pods, `localhost:5001` does obviously not lead to this registry. Hence, we
+	// have to replace it with `garden.local.gardener.cloud:5001` which allows accessing the registry from both local
+	// machine and machine pods.
+	image = strings.ReplaceAll(image, "localhost:5001", "garden.local.gardener.cloud:5001")
```

Comment:
What is this code? We usually have code specific to the local development setup separate from the production code.

Reply:
We need a way to overwrite the image reference to make all this work locally. We can either live with this single line here, or introduce an environment variable that overwrites the image ref and set in the local setup. However, it's not as trivial as we explicitly need to replace `localhost:5001`, so I wouldn't know how to achieve this with this approach. Maybe the most pragmatic way forward for now is to accept this single line as a special exception?

Reply:
As long as we agree that this should be the exception to the general rule I have no problem with this.